strings-csharp/HexToByteArray/HexToByteArray/ConvertHex.cs

```diff
@@ -0,0 +1,35 @@
+﻿namespace HexToByteArray
+{
+    public class ConvertHex
+    {
+        public byte[] FromHexString(string hex)
+        {
+            try
+            {
+                return Convert.FromHexString(hex);
+            }
+            catch (Exception)
+            {
+
+                throw;
+            }
+        }
+
+        public byte[] FromHexStringAlternative(string hex)
```

Comment:
I did a simple test with a 64 character hex string (32 byte resultant array), and by making just a couple of those small changes the code will perform twice as fast and drop memory allocations from 1K to < 60 bytes. Here is the code (note, I also removed the try/catch)

`public byte[] FromHexStringAlternativeUpdated(ReadOnlySpan<char> hex)`
`{`
`        int length = hex.Length;`
`        byte[] bytes = GC.AllocateUninitializedArray<byte>(length >> 1);`
`        for (int i = 0; i < length; i += 2)`
`            bytes[i >> 1] = byte.Parse(hex.Slice(i, 2),`
`                System.Globalization.NumberStyles.HexNumber);`
`        return bytes;`
 `   }`

Here is the benchmark results:
|                          Method |     Mean |    Error |   StdDev |   Gen0 | Allocated |
|-------------------------------- |---------:|---------:|---------:|-------:|----------:|
|        FromHexStringAlternative | 675.0 ns | 12.05 ns | 16.89 ns | 0.5159 |    1080 B |
| FromHexStringAlternativeUpdated | 338.8 ns |  1.17 ns |  1.09 ns | 0.0267 |      56 B |

Reply:
Done. Thanks!